From 56480d70b1384e11f64fa4e13d6237a0cbb5d691 Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Fri, 12 Aug 2016 11:27:45 +0700
Subject: [PATCH 37/42] gst145-omxvideodec: Align a memory size specified when
 exporting a dmabuf file descriptor to the page boundary

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index 033a39c..ddc28a8 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -24,6 +24,7 @@
 #include "config.h"
 #endif
 
+#include <unistd.h>             /* getpagesize() */
 #include "gstomxbufferpool.h"
 #include "gstomxvideodec.h"
 
@@ -648,14 +649,17 @@ gst_omx_buffer_pool_acquire_buffer (GstBufferPool * bpool,
           OMXR_MC_VIDEO_DECODERESULTTYPE *decode_res =
               (OMXR_MC_VIDEO_DECODERESULTTYPE *) omx_buf->
               omx_buf->pOutputPortPrivate;
+          gint page_size;
   
           phys_addr = (guint) decode_res->pvPhysImageAddressY + vmeta->offset[i];
           plane_size[i] = vmeta->stride[i] *
               GST_VIDEO_INFO_COMP_HEIGHT (&pool->video_info, i);
+          page_size = getpagesize ();
   
           res =
               mmngr_export_start_in_user (&vdbuf_data->id_export[i],
-              plane_size[i], (unsigned long) phys_addr, &dmabuf_fd[i]);
+              (plane_size[i] + page_size - 1) & ~(page_size - 1),
+              (unsigned long) phys_addr, &dmabuf_fd[i]);
           if (res != R_MM_OK) {
             GST_ERROR_OBJECT (pool,
                 "mmngr_export_start_in_user failed (phys_addr:0x%08x)",
-- 
1.9.1

